docs/src/apireference.md

```diff
@@ -316,6 +316,12 @@ be thrown:
 InvalidIndex
 ```
 
+As discussed in [JuMP mapping](@ref), for scalar constraint with a nonzero
+function constant, a [`ScalarFunctionConstantNotZero`](@ref) should be thrown:
```

Comment:
From this statement and the test below, does this mean that solvers are obligated to throw an error in this case? This is a pretty big shift in design and needs a bit more discussion.

@odow @joaquimg

Reply:
I wanted to write `may`, in other places, I write `may` IIRC